extmod/uasyncio/stream.py

```diff
@@ -40,7 +40,15 @@ async def readline(self):
                 return l
 
     def write(self, buf):
-        self.out_buf += buf
+        if self.out_buf == b'':
+            ret = self.s.write(buf)
+            if ret is None:
+                ret = 0
+            elif ret == len(buf):
+                return
+            self.out_buf = buf[ret:] # must make a copy
+        else:
+            self.out_buf += buf
 
     async def drain(self):
         mv = memoryview(self.out_buf)
```

Comment:
this could now return immediately if `out_buf` is empty (without creating a memoryview)